src/XMakeBuildEngine/Xml/ProjectXmlUtilities.cs

```diff
@@ -94,14 +94,31 @@ internal static void ThrowProjectInvalidChildElement(string name, string parentN
         /// </summary>
         internal static void VerifyThrowProjectValidNamespace(XmlElementWithLocation element)
         {
-            if (element.Prefix.Length > 0 ||
-                !String.Equals(element.NamespaceURI, XMakeAttributes.defaultXmlNamespace, StringComparison.OrdinalIgnoreCase))
+            // If a namespace was specified it must be the default MSBuild namespace.
+            if (!VerifyValidProjectNamespace(element))
             {
-                ProjectErrorUtilities.ThrowInvalidProject(element.Location, "CustomNamespaceNotAllowedOnThisChildElement", element.Name, element.ParentNode.Name);
+                ProjectErrorUtilities.ThrowInvalidProject(element.Location,
+                    "CustomNamespaceNotAllowedOnThisChildElement", element.Name, element.ParentNode?.Name);
             }
         }
 
         /// <summary>
+        /// Verify that if a namespace is specified it matches the default MSBuild namespace.
+        /// </summary>
+        /// <param name="element">Element to check namespace.</param>
+        /// <returns>True when the namespace is in the MSBuild namespace or no namespace.</returns>
+        internal static bool VerifyValidProjectNamespace(XmlElementWithLocation element)
```

Comment:
I extracted this out to another method. Agreed it was not straightforward as it was, and it was also duplicated. This look better?